tests/init_environment.sh

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+set -x
+set -e
```

Comment:
I'm not totally sure I follow -- I don't think any of the tests will succeed if the compilation test fails, no?  The steps we're performing in this script are really just setting up the environment -- if somebody posts a PR that can't be compiled; that's a pretty serious problem that would prevent further tests from succeeding anyway.

Reply:
Indicating that something failed is good. But it's also important to indicate why it failed.

Let's see this example:

```
    - test -z $(find -name '*.pyc' -o -name '*.pyo')
    - python -tt -m compileall -f bCNC
```

First test makes sure that you didn't accidentaly commited .pyc file.
Second test tries if all .py files can be compiled

If you have both .pyc file and syntax error commited, you will be only notified about .pyc file and then execution of `init_environment.sh` will stop. So you will delete .pyc files, commit, push and wait for travis build again, only to get compilation error.

On the other hand, when the test is in `.travis.yml`, it will run all tests even if some of them fail. That way you will know all problems at once. You will see that there are unwanted .pyc files and you have some syntax error in .py, so you can fix that in single commit.

Reply:
It's like when evil teacher will refuse to accept your homework and gives you list of stuff you have to fix. Then you came back with fixed homework and suddenly he mades up new problems that he didn't mentioned during first trial. I hate this.

Reply:
I do understand what you're describing, and would totally agree if the things we were discussing were actual likely things.  Honestly, though, I work on a ton of opensource projects, and I'm not sure I've ever seen somebody push a PR including a `.pyc` file (they are traditionally, and in this project, in the `.gitignore` file), and pushing-up code that isn't parseable is pretty rare.  That said, most of the projects I work on have rather better test coverage, so maybe that helps matters.  In either case, you're totally welcome to make whatever changes you want on this.